packages/theme-a11y/theme-a11y.puppeteer.test.js

```diff
@@ -0,0 +1,304 @@
+      <button id="disabledButton" disabled></button>
+    `);
+
+    const valid = ["button"];
+    const invalid = ["disabledButton"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects enabled, non-hidden input elements", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <input type="text" id="input"></input>
+      <input type="hidden" id="hiddenInput"></input>
+    `);
+
+    const valid = ["input"];
+    const invalid = ["hiddenInput"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects valid <object> elements", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <object id="object" data="movie.swf" type="application/x-shockwave-flash" />
+    `);
+
+    const valid = ["object"];
+    const invalid = [""];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects enabled <select> elements", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <select id="validSelect">
+        <option val="1">1</option>
+      </select>
+      <select id="disabledSelect" disabled>
+        <option val="1">1</option>
+      </select>
+    `);
+
+    const valid = ["validSelect"];
+    const invalid = ["disabledSelect"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects enabled <textarea> elements", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <textarea id="validTextArea"></textarea>
+      <textarea id="disabledTextArea" disabled></textarea>
+    `);
+
+    const valid = ["validTextArea"];
+    const invalid = ["disabledTextArea"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects elements with tabindex attribute", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <div id="withTabIndex" tabindex="-1"></div>
+      <div id="withoutTabIndex"></div>
+    `);
+
+    const valid = ["withTabIndex"];
+    const invalid = ["withoutTabIndex"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects elements with draggable attribute", async () => {
+    await page.setContent(`
+      <script type="module">
+        window.focusable = ${focusable.toString()};
+      </script>
+      <div id="draggableTrue" draggable="true"></div>
+      <div id="draggableFalse" draggable="true"></div>
+      <div id="withoutDraggable"></div>
+    `);
+
+    const valid = ["draggableTrue"];
+    const invalid = ["draggableFalse", "withoutDraggable"];
+
+    await expectElements(valid, invalid);
+  });
+
+  test("selects only elements which are visible", async () => {
```

Comment:
@huguestennier does this test cover what's needed for visibility?

Reply:
Yup! Even tested when the parent is hidden but contains focusable elements like this : 

```HTML
<div style="display: none;">
  <button>Focusable but hidden button</button>
  <button>Focusable but hidden button</button>
</div>
```
and everything is ✅